### ISSUE TITLE
ambiguity generated by default `copyto!` method

### ISSUE THREAD

**Issue by longemen3000** [JuliaArrays/StaticArrays.jl#1097]
this line:
https://github.com/JuliaArrays/StaticArrays.jl/blob/17e8c377bf959899fb756ba349e17fc344a34bc9/src/broadcast.jl#L65
seems that _copyto! only special handles methods that have static axes. is is correct to assume that those axes only happen on `x <: AbstractArray` ?. if that is the case, then other type trying to dispatch on `copyto!` only needs to define:
```
copyto!(dest::MyType,bc::Base.Broadcsast.AbstractArrayType{N})
```
then, if an external package needs to define 
but the catchall method causes ambiguities in that regard.

I did a test run of StaticArrays on julia 1.8/windows, and removed that method. all tests passed.

**Comment by mateuszbaran** [comment]
I don't understand, what exactly is the problem with this method? What is `x`? Could you give a concrete example of an ambiguity?

> I did a test run of StaticArrays on julia 1.8/windows, and removed that method. all tests passed.

There would very likely be a performance regression, you'd have to benchmark a few statically sized broadcasts to verify that.

**Comment by longemen3000** [comment]
> I don't understand, what exactly is the problem with this method? What is `x`? Could you give a concrete example of an ambiguity?

```
using StaticArrays, DataFrames
df = DataFrame(A=1:3)
tup = (2,4,6)
sa = SVector(tup)
df .= tup #works
df .= sa #ambiguous method
```

**Comment by mateuszbaran** [comment]
I see now, that's actually a good point -- I agree, removing that methods looks like a good idea.

**Comment by mateuszbaran** [comment]
Fixed by #1098 .
